models/crypto/products.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * This file contains wrapper functions to interact with the DB.
+ * This will contain the DB schema if we start consuming an ORM for managing the DB operations
+ */
+
+const logger = require('../../utils/logger')
+const firestore = require('../../utils/firestore')
+const productsModel = firestore.collection('crypto/shop/products')
+
+/**
+ * Fetches the data of products in crypto site
+ * @return {Promise<productsModel|Array>}
+ */
+
+const fetchProducts = async () => {
+  try {
+    const snapshot = await productsModel.get()
+
+    const allProducts = []
+
+    snapshot.forEach((doc) => {
+      allProducts.push({
+        id: doc.id,
+        ...doc.data()
+      })
+    }
+    )
+
+    return allProducts
+  } catch (err) {
+    logger.error('Error retrieving products data', err)
+    throw err
+  }
+}
+
+/**
+ * Fetches the data of a product in crypto site
+ * @return {Promise<productsModel|Object>}
+ */
+
+const fetchProduct = async (id) => {
+  try {
+    const product = await productsModel.doc(id).get()
+    if (product.exists) {
+      return {
+        product: {
+          id: id,
+          ...product.data()
+        }
+      }
+    }
+    return ''
```

Comment:
This line of code is not reachable if the user exists it'll return otherwise promise will be rejected with an error.

Can we refactor this part of the code?
Please let me know if I am missing something.

Reply:
If the user does not exist then it will return undefined